src/api/cddl/base.cddl

```diff
@@ -0,0 +1,36 @@
+blockNo = word64
+epochNo = word64
+slotNo = word64
+
+coin = word64
+
+rational = [int, int]
+
+keyhash = bstr .size 28
+hash = bstr .size 32
+
+relativeTime = int
+
+; Words
+word8  = uint .size 1; 1 byte
+word16 = uint .size 2; 2 bytes
+word32 = uint .size 4; 4 bytes
+word64 = uint .size 8; 8 bytes
+
+telescope7<byron, shelley, allegra, mary, alonzo, babbage, conway>
+  = [pastEra, pastEra, pastEra, pastEra, pastEra, pastEra, currentEra<conway>] /
+    [pastEra, pastEra, pastEra, pastEra, pastEra, currentEra<babbage>] /
+    [pastEra, pastEra, pastEra, pastEra, currentEra<alonzo>] /
+    [pastEra, pastEra, pastEra, currentEra<mary>] /
+    [pastEra, pastEra, currentEra<allegra>] /
+    [pastEra, currentEra<shelley>] /
+    [currentEra<byron>]
+
+ns7<byron, shelley, allegra, mary, alonzo, babbage, conway>
+  = [6, conway] /
+    [5, babbage] /
+    [4, alonzo] /
+    [3, mary] /
+    [2, allegra] /
+    [1, shelley] /
+    [0, byron]
```

Comment:
Can we make sure we have a new line here for POSIX compliance? I imagine a lot of these CDDL files will be processed by other software, good to make sure we don't break anything on that front.